src/andromede/simulation/linear_expression.py

```diff
@@ -170,92 +111,63 @@ def is_zero(self) -> bool:
     def __str__(self) -> str:
         # Useful for debugging tests
         result = _str_for_coeff(self.coefficient) + str(self.variable_name)
-        result += _str_for_time_expansion(self.time_expansion)
-        if self.scenario_operator is not None:
-            result += f".{str(self.scenario_operator)}"
+        return result
+
+    def __repr__(self) -> str:
+        # Useful for debugging tests
+        result = (
+            f"{_str_for_coeff(self.coefficient)}{self.component_id}.{self.variable_name}"
+            f"[{self.time_index},{self.scenario_index}]"
+        )
         return result
 
 
 def generate_key(term: Term) -> TermKey:
     return TermKey(
         term.component_id,
         term.variable_name,
-        term.time_expansion,
-        term.scenario_operator,
+        term.time_index,
+        term.scenario_index,
     )
 
 
 def _merge_dicts(
     lhs: Dict[TermKey, Term],
     rhs: Dict[TermKey, Term],
     merge_func: Callable[[Term, Term], Term],
-    neutral: float,
 ) -> Dict[TermKey, Term]:
     res = {}
     for k, v in lhs.items():
-        res[k] = merge_func(
-            v,
-            rhs.get(
-                k,
-                Term(
-                    neutral,
-                    v.component_id,
-                    v.variable_name,
-                    v.structure,
-                    v.time_expansion,
-                    v.scenario_operator,
-                ),
-            ),
-        )
-    for k, v in rhs.items():
-        if k not in lhs:
+        if k not in rhs:
+            res[k] = v
+        else:
             res[k] = merge_func(
-                Term(
-                    neutral,
-                    v.component_id,
-                    v.variable_name,
-                    v.structure,
-                    v.time_expansion,
-                    v.scenario_operator,
-                ),
                 v,
+                rhs[k],
             )
+    for k, v in rhs.items():
+        if k not in lhs:
+            res[k] = v
     return res
 
 
-def _merge_is_possible(lhs: Term, rhs: Term) -> None:
-    if lhs.component_id != rhs.component_id or lhs.variable_name != rhs.variable_name:
-        raise ValueError("Cannot merge terms for different variables")
-    if (
-        lhs.time_expansion != rhs.time_expansion
-        or lhs.scenario_operator != rhs.scenario_operator
-    ):
-        raise ValueError("Cannot merge terms with different operators")
-    if lhs.structure != rhs.structure:
-        raise ValueError("Cannot merge terms with different structures")
-
-
 def _add_terms(lhs: Term, rhs: Term) -> Term:
-    _merge_is_possible(lhs, rhs)
     return Term(
```

Comment:
Where do you check that the merge is done on terms representing the same variable ?

Reply:
We can consider that we handle this matter upstream in `_merge_dicts`:
since it's a private function, we have complete control over its use so it's not necessary to double check here (as opposed to a public method where we need to to more input validation).